pkgs/games/ja2-stracciatella/default.nix

```diff
@@ -1,45 +1,66 @@
     repo = "ja2-stracciatella";
     rev = "v${version}";
-    sha256 = "1pyn23syg70kiyfbs3pdlq0ixd2bxhncbamnic43rym3dmd52m29";
-  };
-  lockfile = ./Cargo.lock;
-  libstracciatellaSrc = stdenv.mkDerivation {
-    name = "libstracciatella-${version}-src";
-    src = "${src}/rust";
-    installPhase = ''
-      mkdir -p $out
-      cp -R ./* $out/
-      cp ${lockfile} $out/Cargo.lock
-    '';
+    sha256 = "0m6rvgkba29jy3yq5hs1sn26mwrjl6mamqnv4plrid5fqaivhn6j";
   };
   libstracciatella = rustPlatform.buildRustPackage {
     pname = "libstracciatella";
     inherit version;
-    src = libstracciatellaSrc;
-    cargoSha256 = "15djs4xaz4y1hpfyfqxdgdasxr0b5idy9i5a7c8cmh0jkxjv8bqc";
-    doCheck = false;
+    src = "${src}/rust";
+    cargoSha256 = "0blb971cv9k6c460mwq3zq8vih687bdnb39b9gph1hr90pxjviba";
+
+    preBuild = ''
+      mkdir -p $out/include/stracciatella
+      export HEADER_LOCATION=$out/include/stracciatella/stracciatella.h
+    '';
+  };
+  stringTheoryUrl = https://github.com/zrax/string_theory/archive/3.1.tar.gz;
+  stringTheory = fetchurl {
+    url = stringTheoryUrl;
+    sha256 = "1flq26kkvx2m1yd38ldcq2k046yqw07jahms8a6614m924bmbv41";
+  };
+  gtestUrl = https://github.com/google/googletest/archive/release-1.10.0.tar.gz;
+  gtest = fetchurl {
+    url = gtestUrl;
+    sha256 = "1jvdf56gcvhyi9fj70gcigxva2sad6dfmnj3grxfql8mk9x1bjcx";
   };
 in
 stdenv.mkDerivation {
   pname = "ja2-stracciatella";
   inherit src;
   inherit version;
 
-  nativeBuildInputs = [ cmake ];
-  buildInputs = [ SDL2 fltk boost ];
-
-  patches = [
-    ./remove-rust-buildstep.patch
-  ];
+  nativeBuildInputs = [ cmake python ];
+  buildInputs = [ SDL2 fltk rapidjson ] ++ lib.optionals stdenv.isDarwin (with darwin.apple_sdk.frameworks; [ Carbon Cocoa ]);
 
   preConfigure = ''
-    sed -i -e 's|rust-stracciatella|${libstracciatella}/lib/libstracciatella.so|g' CMakeLists.txt
-    cmakeFlagsArray+=("-DEXTRA_DATA_DIR=$out/share/ja2")
+    # Patch to use rust library built with nix
+    sed -i -e 's|add_subdirectory("''${CMAKE_CURRENT_SOURCE_DIR}/dependencies/lib-stracciatella")||g' CMakeLists.txt
+    sed -i -e 's|add_dependencies(''${JA2_BINARY} stracciatella)||g' CMakeLists.txt
+    sed -i -e 's|add_dependencies(''${LAUNCHER_BINARY} stracciatella)||g' CMakeLists.txt
+    sed -i -e 's|''${STRACCIATELLA_LIBRARIES}|${libstracciatella}/lib/libstracciatella_c_api.a dl pthread|g' CMakeLists.txt
+    sed -i -e 's|set(JA2_INCLUDES "")|set(JA2_INCLUDES "${libstracciatella}/include/stracciatella")|g' CMakeLists.txt
+    sed -i -e 's|''${CMAKE_BINARY_DIR}/lib-stracciatella/bin/ja2-resource-pack''${CMAKE_EXECUTABLE_SUFFIX}|${libstracciatella}/bin/ja2-resource-pack|g' CMakeLists.txt
+
+
+    # Patch dependencies that are usually loaded by url
+    sed -i -e 's|${stringTheoryUrl}|file://${stringTheory}|g' dependencies/lib-string_theory/builder/CMakeLists.txt.in
+    sed -i -e 's|${gtestUrl}|file://${gtest}|g' dependencies/lib-gtest/getter/CMakeLists.txt.in
```

Comment:
I think a patch would be preferred here because it does not break that easily and would be less complex.